packages/nextjs/utils/grants.ts

```diff
@@ -7,3 +7,32 @@ export const PROPOSAL_STATUS = {
 } as const;
 
 export type ProposalStatusType = (typeof PROPOSAL_STATUS)[keyof typeof PROPOSAL_STATUS];
+
+export function formatDateFromNow(timestamp: number) {
+  const timestampDate = new Date(timestamp);
+  const now = new Date().getTime();
+  const delta = now - timestampDate.getTime();
+  // Number of days calculated from the given delta.
+  const days = delta / (1000 * 60 * 60 * 24);
+
+  // less than 14 days => n days ago
+  if (days < 14) {
+    const roundedDays = Math.floor(days);
+    return `${roundedDays} day${roundedDays > 1 ? "s" : ""} ago`;
+  }
+  // Less than 8 weeks => n weeks ago
+  else if (days < 8 * 7) {
+    const weeks = Math.floor(days / 7);
+    return `${weeks} week${weeks > 1 ? "s" : ""} ago`;
+  }
+  // Less than 24 months => n months ago
+  else if (days < 24 * 30) {
+    const months = Math.floor(days / 30);
+    return `${months} month${months > 1 ? "s" : ""} ago`;
+  }
+  // More than 24 months => n years ago
+  else {
+    const years = Math.floor(days / 365);
+    return `${years} year${years > 1 ? "s" : ""} ago`;
+  }
+}
```

Comment:
Instead, we could use this from [stackoverflow](https://stackoverflow.com/a/69122877) since it works a bit better and tried/tested by different people : 

<details>
<summary>
TS Code: 
</summary>

```ts
export function formatDateFromNow(input: string | number | Date) {
  const date = input instanceof Date ? input : new Date(input);
  const formatter = new Intl.RelativeTimeFormat("en");
  const ranges = [
    ["years", 3600 * 24 * 365],
    ["months", 3600 * 24 * 30],
    ["weeks", 3600 * 24 * 7],
    ["days", 3600 * 24],
    ["hours", 3600],
    ["minutes", 60],
    ["seconds", 1],
  ] as const;
  const secondsElapsed = (date.getTime() - Date.now()) / 1000;

  for (const [rangeType, rangeVal] of ranges) {
    if (rangeVal < Math.abs(secondsElapsed)) {
      const delta = secondsElapsed / rangeVal;
      return formatter.format(Math.round(delta), rangeType);
    }
  }
}
```
</details>

Because your code converts 19, 15 days ago => 2 weeks

And the StackOverflow code converts 19 days => 3 weeks ago and 15 day => 2 weeks ago

Also maybe we can move this function to `utils/index.ts` and export this from their and also more common ingenreal utils in it